### ISSUE TITLE
wxDebugReport doesn't work if a fatal exception happens in a detached or joined std::thread

### ISSUE THREAD

**Issue by rossanoparis** [wxWidgets/wxWidgets#23958]
### To Reproduce
To reproduce the issue I modified the `debugrpt.cpp` file contained in "debugrpt" example.
The function I modified to reproduce the issue is: `MyFrame::OnReportForCrash`
![image](https://github.com/wxWidgets/wxWidgets/assets/33904565/f4ce96ef-1bac-40a7-a9d9-b76450b0d3cb)

Here you can find the modified `debugrpt.cpp` file.
[debugrpt.zip](https://github.com/wxWidgets/wxWidgets/files/12878703/debugrpt.zip)

### Description

Perhaps I'm missing something regarding `wxDebugReport` and fatal exception handling in general.
I couldn'd find information for `wxDebugReport, OnFatalException(), wxHandleFatalExceptions()` related to code executed in other thread other than the main one.
The example provided with wx, works fine with all exceptions generated in the main thread, but using a detached or joined std::thread the example behaviour is not the same; the function `OnFatalException()` is not called at all.

Another topic which is not clear to me is the following one.
The function `OnFatalException()` calls `MyApp::GenerateReport()` and inside this function there is a call to `wxDebugReportPreviewStd().Show(*report)`
My question is: calling the `Show()` method, shouldn't lead in a second exception if the calling thread is not the main one?

### Platform and version information

- wxWidgets version  you use: 3.2.2.1
- wxWidgets port  you use: wxMSW, wxGTK
- OS  and its version: Windows 11 Pro 10.0.22621 Build 22621, Linux 5.4.0-150-generic
 

+ GTK version:  3.22.30
+ Which GDK  backend is used: X11

- Visual Studio version: 17.7.4
- GCC version: 11.4.0

**Comment by vadz** [comment]
`wxDebugReport` itself can only be used from the main thread, but ideally we would indeed call `OnFatalException()` even for the exceptions from the other threads. It would probably be enough to call `DisableAutomaticSETranslator()` (which is a bad name and not wx-prefixed neither...) from the thread function to make it happen.

As for `wxHandleFatalExceptions()` you should already be able to call it from any thread you create.

**Comment by rossanoparis** [comment]
Thank you @vadz for your reply.

Excuse me, but I need a clarification before starting to test.
The two functions you mentioned, `DisableAutomaticSETranslator()` and `wxHandleFatalExceptions()` should be called continuosly inside the secondary threads or it is necessary to call them one time only.

**Comment by rossanoparis** [comment]
Thank you @vadz

Based on my tests, it's not necessary to call the function `DisableAutomaticSETranslator()` and even calling  'wxHandleFatalExceptions()` seems not necesary for each `std::thread`

I call 'wxHandleFatalExceptions()` on application Initialization only and for each function called using a `std:thread` its code is included in a try catch.

In this way on MSW the callback `OnFatalException()` is called.
Using Linux, seems not necessary to use this way; it works as it is.

Here is an example.
```
#ifdef _WIN32
	#include <wx/msw/seh.h>

	#define wxhTRY wxSEH_TRY
	#define wxhCATCH wxSEH_HANDLE(THREAD_ERROR_EXIT)
#else
	#define wxhTRY
	#define wxhCATCH
#endif
```

`std::thread(&Class::MyFunction, this).detach();`

```
void Class::MyFunction()
{
   wxhTRY {

     ... my code ...

   } wxhCATCH
}
```

Does it make sense?

**Comment by vadz** [comment]
Using explicit `__try` around the entry function does work, of course, the idea behind `wxHandleFatalExceptions()` is that it would work even without them...

**Comment by rossanoparis** [comment]
Yes, you are right of course, but actually MSW porting doesn't behave like that.
I had to find a temporary work-around.
Do you think it could be fixed?

**Comment by vadz** [comment]
We could do it for `wxThread`s but I don't know how to do it for arbitrary other threads, we'd need to find some way to execute a hook on each thread startup.

Maybe we need `wxInitializeThread()` or something like this? It could be useful for other things...

**Comment by rossanoparis** [comment]
I suppose wxThread should be OK, as I get the work-around I found looking at its code under windows source.
Perhaps waht it make sense is a note on documentation regarding the use of `std::thread` for a windows application.
For Linux it is not necessary as it works as expected even using `std::thread`
